### PR TITLE
feat: add ability to create public read instance

### PR DESCRIPTION
**Pull request by rayanebel**
## Description

The purpose of this PR is to add all the required options to allow us to create public read instances.

### CODE REVIEW

**Comment by rayanebel** [comment]
cc @imrannayer I've updated the docs, I think that's why the CI was failing previously

File: main.tf [hunk 1]
```diff
@@ -159,9 +159,9 @@ resource "google_alloydb_instance" "primary" {
     content {
       enable_public_ip = var.primary_instance.enable_public_ip
       dynamic "authorized_external_networks" {
-        for_each = var.primary_instance.cidr_range == null ? [] : ["authorized_external_networks"]
+        for_each = var.primary_instance.cidr_range != null ? var.primary_instance.cidr_range : []
```
Comment (imrannayer) [inline main.tf:162]: @rayanebel whats the point of changing this? It is working fine and new code is not changing anything except it will show diff in existing user's resource? Can you plz revert this change?

Comment (rayanebel) [reply main.tf]: @imrannayer The current code is not working when you add a network CIDR. Here the result I have If I revert the code

```bash
 Error: Error creating Instance: googleapi: Error 400: The request was invalid: invalid authorized network CIDR range "authorized_external_networks"
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.BadRequest",
│     "fieldViolations": [
│       {
│         "field": "networkConfig.authorizedExternalNetworks"
│       }
│     ]
│   },
│   {
│     "@type": "type.googleapis.com/google.rpc.RequestInfo",
│     "requestId": "dab977d705ea321c"
│   }
│ ]
│
│   with module.alloy-db.google_alloydb_instance.primary,
│   on .terraform/modules/alloy-db/main.tf line 146, in resource "google_alloydb_instance" "primary":
│  146: resource "google_alloydb_instance" "primary" {
│
╵
```

**Terraform plan**
```bash
 + machine_config           = [
          + {
              + cpu_count = 2
            },
        ]
      + name                     = (known after apply)
      + network_config           = [
          + {
              + authorized_external_networks = [
                  + {
                      + cidr_range = "authorized_external_networks"
                    },
                ]
              + enable_public_ip             = true
            },
        ]
      + psc_instance_config      = []
```

Comment (imrannayer) [reply main.tf:162]: thats cos you have changed line `164`. If you keep line `162` and 164 same, code will work fine. Just revert all the changes you made in `network_config` block.

```
  dynamic "network_config" {
    for_each = var.primary_instance.enable_public_ip ? ["network_config"] : []
    content {
      enable_public_ip = var.primary_instance.enable_public_ip
      dynamic "authorized_external_networks" {
        for_each = var.primary_instance.cidr_range == null ? [] : ["authorized_external_networks"]
        content {
          cidr_range = var.primary_instance.cidr_range
        }
      }
    }
  }
```

File: main.tf [hunk 2]
```diff
@@ -161,7 +161,7 @@ resource "google_alloydb_instance" "primary" {
       dynamic "authorized_external_networks" {
```
Comment (imrannayer) [inline main.tf:161]: @rayanebel I think we have to change this block and make it similar to the  block you added for read instances. 

```
  dynamic "network_config" {
    for_each = var.primary_instance.enable_public_ip ? ["network_config"] : []
    content {
      enable_public_ip = var.primary_instance.enable_public_ip
      dynamic "authorized_external_networks" {
        for_each = var.primary_instance.cidr_range == null ? [] : toset(var.primary_instance.cidr_range)
        content {
          cidr_range = authorized_external_networks.value
        }
      }
    }
  }
```

File: main.tf [hunk 3]
```diff
@@ -217,6 +217,21 @@ resource "google_alloydb_instance" "read_pool" {
   labels            = var.primary_instance.labels
   annotations       = var.primary_instance.annotations
 
+  dynamic "network_config" {
+    for_each = each.value.enable_public_ip ? ["network_config"] : []
+    content {
+      enable_public_ip = each.value.enable_public_ip
+
+      dynamic "authorized_external_networks" {
+        for_each = each.value.cidr_range != null ? each.value.cidr_range : []
```
Comment (imrannayer) [inline main.tf:226]: @rayanebel I am wondering if we need to change `each.value.cidr_range` to `toset(each.value.cidr_range)` as `cidr_range` is defined as `list` and for_each accepts `set`.

File: main.tf (hunk 2 above)
Comment (rayanebel) [reply main.tf:161]: @imrannayer Done, I've tested in my cluster and its working

File: main.tf (hunk 3 above)
Comment (rayanebel) [reply main.tf]: @imrannayer Done, I've also tested this part and it works

**Review by imrannayer — APPROVED** [review]
